File: calcularnumber.cs
using System;

class Program
{
    // Método que calcula o quadrado de um número
    static int Quadrado(int num)
    {
        int resultado = num * num; // Variável local
        return resultado;
    }

    // Método que calcula o dobro de um número sem modificar o original
    static int DobroSemModificar(int num)
    {
        return num * 2;
    }

    // Método que modifica o valor original usando ref
    static void DobroModificando(ref int num)
    {
        num *= 2;
    }

    // Métodos sobrecarregados para soma
    static int Soma(int a, int b)
    {
        return a + b;
    }

    static int Soma(int a, int b, int c)
    {
        return a + b + c;
    }

    static double Soma(double a, double b)
    {
        return a + b;
    }

    static void Main()
    {
        int num = 5;
        Console.WriteLine($"Quadrado de {num}: {Quadrado(num)}");
        
        Console.WriteLine($"Dobro de {num} sem modificar: {DobroSemModificar(num)}");
        
        DobroModificando(ref num);
        Console.WriteLine($"Dobro de {num} modificando: {num}");
        
        Console.WriteLine($"Soma de 2 e 3: {Soma(2, 3)}");
        Console.WriteLine($"Soma de 1, 2 e 3: {Soma(1, 2, 3)}");
        Console.WriteLine($"Soma de 1.5 e 2.5: {Soma(1.5, 2.5)}");
    }
}

File: calcularnumber.c
#include <stdio.h>

// Função que calcula o quadrado de um número
int quadrado(int num) {
    int resultado = num * num; // Variável local
    return resultado;
}

// Função que calcula o dobro de um número sem modificar o original
int dobro_sem_modificar(int num) {
    return num * 2;
}

// Função que modifica o valor original usando ponteiro
void dobro_modificando(int *num) {
    *num *= 2;
}

// Funções sobrecarregadas para soma
int soma(int a, int b) {
    return a + b;
}

int soma3(int a, int b, int c) {
    return a + b + c;
}

double soma_double(double a, double b) {
    return a + b;
}

int main() {
    int num = 5;
    printf("Quadrado de %d: %d\n", num, quadrado(num));

    printf("Dobro de %d sem modificar: %d\n", num, dobro_sem_modificar(num));
    
    dobro_modificando(&num);
    printf("Dobro de %d modificando: %d\n", num, num);

    printf("Soma de 2 e 3: %d\n", soma(2, 3));
    printf("Soma de 1, 2 e 3: %d\n", soma3(1, 2, 3));
    printf("Soma de 1.5 e 2.5: %.2f\n", soma_double(1.5, 2.5));
    
    return 0;
}
